lib/appengine.dart

```diff
@@ -41,10 +41,24 @@ const Symbol _APPENGINE_CONTEXT = #appengine.context;
 /// You can provide a [port] if you want to run the HTTP server on a different
 /// port than the `8080` default.
 ///
+/// The optional argument [shared] argument specifies whether additional
+/// AppEngine servers can bind to the same `port`. If `shared` is `true` and
+/// more AppEngine servers from this isolate or other isolates are bound to the
+/// port, then the incoming connections will be distributed among all the bound
+/// servers. Connections can be distributed over multiple isolates this way.
+///
+/// The optional [onAcceptingConnections] callback, if provided, will be
+/// notified when the server is accepting connections on [port].
+///
 /// The returned `Future` will complete when the HTTP server has been shutdown
 /// and is no longer serving requests.
-Future runAppEngine(void handler(HttpRequest request),
-    {Function onError, int port = 8080, bool shared = false}) {
+Future runAppEngine(
+  void handler(HttpRequest request), {
+  Function onError,
+  int port = 8080,
+  bool shared = false,
+  void onAcceptingConnections(),
```

Comment:
Should we make this callback take an `int` parameter that specifies the port? That seems useful if `port` were specified as `0` and you want to know what port was selected.  I'm not sure if that's even allowed on AppEngine or not though.

Reply:
Documentation says that "appengine flex custom runtimes" will route requests to port `8080`.
https://cloud.google.com/appengine/docs/flexible/custom-runtimes/build#listening_to_port_8080

I suppose it's possible to run this locally, and then have it listen on a different port.

Reply:
Updated to pass the address and the port.  And yes, this information is useful is local development.